siliconcompiler/tools/vpr/place.py

```diff
@@ -24,6 +26,39 @@ def setup(chip, clobber=True):
     chip.add('tool', tool, 'task', task, 'output', design + '.place', step=step, index=index)
 
 
+################################
+# Pre_process (pre executable)
+################################
+
+
+def pre_process(chip):
+    ''' Tool specific function to run before step execution
+    '''
+
+    design = chip.top()
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+
+    if 'pins' in chip.getkeys('input', 'constraint'):
+        chip.logger.info("Using pin constraints file instead of pin constraints from chip.set")
+    else:
+        all_component_constraints = chip.getkeys('constraint', 'component')
+        all_place_constraints = {}
+        if (len(all_component_constraints) > 0):
```

Comment:
this check isn't needed, if all_component_constraint is empty then the for loop will no nothing and you can just check the dict before calling generate_vpr_constraints_xml